### ISSUE TITLE
WKWebView - http://localhost same origin policy

### ISSUE THREAD

**Issue by philipp-schaerer-lambdait** [ionic-team/capacitor#788]
With ionic capacitor the used webview is fixed to WKWebView. This leads to the fact that the App is served internally via http://localhost if i understood this right.
We did face some CORS issues, but so far we were able to solve them by configuring the servers which were owned or using JSONP API in case of the external Instagram API that was being used.

But we are facing now some issues regarding Google Ads, they get integrated via dynamic iframes, until now this wasn't an issue. The iframes created by the ads seem to have some workarounds regarding Same Origin Policy (SOP). But lately a lot of the shown ads had shown some SOP error messages and the ads couldn't be displayed correctly anymore.
```
Blocked a frame with origin "http://localhost:8237" from accessing a frame with origin "http://tpc.googlesyndication.com". Protocols, domains, and ports must match.
```

Are there some workarounds or configurations using the WKWebView that can be used to allow those iframes and requests? Or could it be possible to get rid of the localhost origin serving?

Some interesting WKWebView options regarding this topic that some one did point out:
https://stackoverflow.com/questions/36013645/setting-disable-web-security-and-allow-file-access-from-files-in-ios-wkwebvi

**Comment by jcesarmobile** [comment]
we won't change the `http://` access to `file://` access, file protocol has a few problems on WKWebView that can only be fixed with the local webserver.
Also the local webserver is the better option for the angular base href.

Does this happen with Ionic webview or only with Capacitor?

**Comment by philipp-schaerer-lambdait** [comment]
I've testet it only with Capacitor, I'll try to test it without capacitor but using WKWebView but that can take a while, I'll update this as soon as I got more information.

**Comment by nklayman** [comment]
On android, I get a Cross-Origin Read Blocking notice whenever I try to set the `referer` header on a network request. The request itself fails due to the `Access-Control-Allow-Origin` not being on the server. It works fine on cordova (likely because app is loaded via `file://` protocol).

**Comment by jcesarmobile** [comment]
curious if this is still an issue on latest beta that uses `capacitor://` scheme instead of http, @philipp-schaerer-lambdait are you still using Capacitor? can you check with latest beta?

**Comment by ryaa** [comment]
I checked with the latest beta.13 and this is still problem on both iOS and Android. I noticed that on iOS the scheme to load the app resources is capacitor://localhost but on Android this is still http://localhost.

On Android the CORS error is as follows:
`
Access to XMLHttpRequest at '<API_URL>' from origin 'http://localhost' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.
`

**Comment by 3alampro** [comment]
CORS issues still exist even after upgrading to the latest release to use `capacitor://` to fix it we have to configure our server or backend to allow for it but some cases we don't have access to change server or service configs when using NativeScript with angular or Vue we don't run into any CORS issues I hope this fixed ASAP since we love using ionic and capacitor than any other solutions out there

**Comment by gabriele-sacchi** [comment]
Any help on this related question please?  :) https://stackoverflow.com/questions/56044902/how-to-add-missing-referrer-header-to-cordova-ios-app

**Comment by jcesarmobile** [comment]
Closing since Capacitor no longer uses the local web server and uses a custom scheme to serve the app content, so the OP is out of date.

If you still face a similar issue with latest version, please, report a new issue providing more information.

Note to other people who have commented, this is related to iframes. For XHR/fetch calls to web servers you need to configure them to allow CORS, that's a WKWebView security feature and there is nothing we can do about it.

**Comment by scotch83** [comment]
> Closing since Capacitor no longer uses the local web server and uses a custom scheme to serve the app content, so the OP is out of date.
> 
> If you still face a similar issue with latest version, please, report a new issue providing more information.
> 
> Note to other people who have commented, this is related to iframes. For XHR/fetch calls to web servers you need to configure them to allow CORS, that's a WKWebView security feature and there is nothing we can do about it.

Sorry @jcesarmobile can you please let know where we can find documentation on the origin used by Capacitor to perform requests?

Thank you
